buildlog/buildlog_test.go

```diff
@@ -0,0 +1,201 @@
+package buildlog
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type writeCloser struct {
+	io.Writer
+}
+
+func (wc writeCloser) Close() error {
+	return nil
+}
+
+func write(tb testing.TB, filepath string, inputs []string) {
+	tb.Helper()
+
+	for _, input := range inputs {
+		file, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		_, err = file.Write([]byte(input))
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		err = file.Close()
+		if err != nil {
+			tb.Fatal(err)
+		}
+	}
+}
+
+func TestRun(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		tmpFile, err := ioutil.TempFile("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer tmpFile.Close()
+
+		testInputs := []string{
+			`{"t": 1580198209, "m": "test 1", "n": 0, "s": "main"}` + "\n",
+			`{"t": 1580198222, "m": "test 2", "n": 1, "s": "main"}` + "\n",
+		}
+		go write(t, tmpFile.Name(), testInputs)
+
+		parent, cancel := context.WithCancel(context.Background())
+		writer := bytes.NewBuffer(nil)
+		l := log{
+			file:   tmpFile,
+			writer: writer,
+			ctx:    parent,
+			cancel: cancel,
+		}
+
+		go l.Run()
+
+		time.Sleep(3 * time.Second)
+		l.Stop()
+
+		expected := []string{
+			"2020-01-28 16:56:49 +0900 JST: test 1\n",
+			"2020-01-28 16:57:02 +0900 JST: test 2\n",
+		}
+		assert.Equal(t, strings.Join(expected, ""), writer.String())
+	})
+
+	t.Run("failure by parsing error", func(t *testing.T) {
+		tmpFile, err := ioutil.TempFile("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer tmpFile.Close()
+
+		testInputs := []string{
+			`{"t": 1580198209, "m": "test 1", "n": 0, "s": "main"}` + "\n",
+			`{"t": 1580198222, "m": "test 2", "n": 1, "s": "main"` + "\n",
```

Comment:
The input value of this test case looks similar with the success case. 
In my opinion, such test cases make us confusing. Failure case should be more simple if we don't want to test JSON parse.

Reply:
I fixed it.